scripts/tools/metricTracer.yml

```diff
@@ -0,0 +1,255 @@
+    - $X_RESULT_NOT_READY: 
+        - "Metric tracer is disabled and no data is available to read."
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Create a metric decoder for a given metric tracer."
+version: "1.10"
+class: $tMetricDecoder
+name: CreateExp
+params:
+    - type: "$t_metric_tracer_exp_handle_t"
+      name: hMetricTracer
+      desc: "[in] handle of the metric tracer"
+    - type: $t_metric_decoder_exp_handle_t*
+      name: phMetricDecoder 
+      desc: "[out] handle of the metric decoder object"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Destroy a metric decoder."
+version: "1.10"
+class: $tMetricDecoder
+name: DestroyExp
+params:
+    - type: $t_metric_decoder_exp_handle_t
+      name: phMetricDecoder 
+      desc: "[in] handle of the metric decoder object"
+--- #--------------------------------------------------------------------------
+type: function
+desc: |
+      Return the list of the decodable metrics that can be collected in the tracer for the which the metric decoder 
+      handle was provided. The decodable metrics handles returned by this API are only valid to decode metrics raw data 
+      with $tMetricTracerDecodeExp(). Decodable metric handles are not valid to compare with metrics handles included in 
+      metric groups."
+version: "1.10"
+class: $tMetricDecoder
+name: GetDecodableMetricsExp
+params:
+    - type: $t_metric_decoder_exp_handle_t
+      name: phMetricDecoder 
+      desc: "[in] handle of the metric decoder object"
+    - type: uint32_t* 
+      name: pCount 
+      desc: |
+            [in,out] pointer to number of decodable metric in the hMetricDecoder handle. If count is zero, then the driver shall 
+            update the value with the total number of decodable metrics available in the decoder. if count is greater than zero 
+            but less than the total number of decodable metrics available in the decoder, then only that number will be returned. 
+            if count is greater than the number of decodable metrics available in the decoder, then the driver shall update the 
+            value with the actual number of decodable metrics available. 
+    - type: $t_metric_handle_t*
+      name: phMetrics 
+      desc: "[in,out] [range(0, pCount)] array of handles of decodable metrics in the hMetricDecoder handle provided."
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Decode raw events collected from a tracer."
+version: "1.10"
+class: $tMetricTracer
+name: DecodeExp
+params:
+    - type: $t_metric_decoder_exp_handle_t
+      name: phMetricDecoder 
+      desc: "[in] handle of the metric decoder object"
+    - type: size_t*
+      name: pRawDataSize
+      desc: | 
+            [in,out] size in bytes of raw data buffer. If pMetricEntriesCount is greater than zero but less than total number of 
```

Comment:
This sentence has to explain pRawDataSize right ?

Reply:
yes, but the text says that if the user submits an limited input buffer (by a fixed pMetricEntriesCount) then pRawDataSize will tell in output how much data was processed.